File: Codeforces/C462B.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    long long n, k;
    cin>>n>>k;
    string str;
    cin>>str;
    long long hash[26] = {};
    for (int i = 0; i < n; i++) {
        hash[str[i] - 'A']++;
    }
    sort(hash, hash+26, greater<int>());
    long long res = 0;
    for (int i = 0; i < 26; i++) {
        if (k > hash[i]) {
            res += (hash[i]*hash[i]);
            k -= hash[i];
        } else {
            res += (k*k);
            break;
        }
    }

    cout<<res;
}

File: Codeforces/C462B.java
import java.util.Scanner;
import java.util.Collections;
import java.util.Arrays;
public class C462B {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n, k;
        n = in.nextInt();
        k = in.nextInt();
        String str;
        str = in.next();
        long [] hash = new long [26];
        for (int i = 0; i < n; i++) {
            hash[(int)(str.charAt(i) - 'A')]++;
        }
        Arrays.sort(hash);
        long res = 0;
        for (int i = 25; i >= 0; i--) {
            if (k > hash[i]) {
                res += ((long)hash[i])*((long)hash[i]);
                k -= hash[i];
            } else {
                res += ((long)k)*((long)k);
                k = 0;
                break;
            }
        }
        System.out.print(res);
    }
}